src/App/Scan/FossaV1.hs

```diff
@@ -0,0 +1,98 @@
+module App.Scan.FossaV1
+  ( uploadAnalysis
+  , UploadResponse(..)
+  , FossaError(..)
+  ) where
+
+import Control.Carrier.Error.Either
+import Data.List (isInfixOf)
+import Data.Text.Encoding (encodeUtf8)
+import Effect.Logger
+import qualified Network.HTTP.Client as HTTP
+import Network.HTTP.Req
+import qualified Network.HTTP.Types as HTTP
+import Prologue
+import Srclib.Converter (toSourceUnit)
+import Srclib.Types
+import Types
+
+-- TODO: git commit?
+cliVersion :: Text
+cliVersion = "spectrometer"
+
+uploadUrl :: Url 'Https
+uploadUrl = https "app.fossa.com" /: "api" /: "builds" /: "custom"
+
+newtype FossaReq m a = FossaReq { runFossaReq :: ErrorC FossaError m a }
+  deriving (Functor, Applicative, Monad, MonadIO)
+
+instance (Algebra sig m, Effect sig, MonadIO m) => MonadHttp (FossaReq m) where
+  handleHttpException = FossaReq . throwError . mangleError
```

Comment:
This is disgusting, but unfortunately the best option for now. Normally, `Req` throws exceptions in IO (i.e., the type doesn't reflect the kinds of exceptions it can throw) -- but we'd like to be explicit and return out `Either FossaError UploadResponse`.

This bit of confusing boilerplate re-throws the exceptions with the `Error FossaError` effect